base_classes/NXdetector.nxdl.xml

```diff
@@ -31,11 +31,17 @@
             >
 
   <symbols>
-    <doc>These symbols will be used below to coordinate datasets with the same shape.</doc>
-    <symbol name="np"><doc>number of scan points (only present in scanning measurements)</doc></symbol>
+    <doc>
+      These symbols will be used below to illustrate the coordination of the rank and sizes of datasets and the 
+      preferred ordering of the dimensions. Each of these are optional (so the rank of the datasets 
+      will vary according to the situation) and the general ordering principle is slowest to fastest.
+      The type of each dimension should follow the order of scan points, detector pixels,
+      then time-of-flight (i.e. spectroscopy, spectrometry).
+    </doc>
+    
+    <symbol name="nP"><doc>number of scan points (only present in scanning measurements)</doc></symbol>
     <symbol name="i"><doc>number of detector pixels in the first (slowest) direction</doc></symbol>
     <symbol name="j"><doc>number of detector pixels in the second (faster) direction</doc></symbol>
-    <symbol name="k"><doc>number of detector pixels in the third (if necessary, fastest) direction</doc></symbol>
```

Comment:
I like the new documentation, but I'm worried about dropping `k` here, since it is [used](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-data-field) in NxMX.    Is the idea behind dropping `k` that it's not used below in the NXdetector base class?  Perhaps in addition to dropping it, a note could be added about how some detectors have a 3rd dimension `k` of pixels that often corresponds to module number?

Reply:
I dropped `k` because it was not used anywhere. else below. The extra text in the doc string points out that the symbols and dimensions ordering, sizes, etc. are simple illustrative, rather than restrictive. I agree that a detector can produce data with more than 2 dimensions, but I don't feel it is so important to say so explicitly in this piece of reference documentation. If we keep `k`, then we should use it, but using it consistently is going to add lots of "noise" so I don't see it being worthwhile.

Reply:
👍